platforms/documentation/docs/build.gradle.kts

```diff
@@ -85,24 +85,47 @@ dependencies {
 
 asciidoctorj {
     setVersion("2.5.13")
-    modules {
-        pdf.use()
-        pdf.setVersion("2.3.10")
-    }
+    modules.pdf.setVersion("2.3.10")
     // TODO: gif are not supported in pdfs, see also https://github.com/gradle/gradle/issues/24193
     // TODO: tables are not handled properly in pdfs
     fatalWarnings.add(
         Regex("^(?!GIF image format not supported|dropping cells from incomplete row detected end of table|.*Asciidoctor PDF does not support table cell content that exceeds the height of a single page).*").toPattern()
     )
 }
 
-tasks.withType<AsciidoctorTask>().configureEach {
-    if (name == "userguideSinglePagePdf") {
-        asciidoctorj.docExtensions(
+tasks.named<AsciidoctorTask>("userguideSinglePagePdf").configure {
+    val task = this
+    task.extensions.getByType<org.asciidoctor.gradle.jvm.AsciidoctorJExtension>().apply {
+        docExtensions(
             project.dependencies.create(project(":docs-asciidoctor-extensions-base"))
         )
-    } else {
-        asciidoctorj.docExtensions(
+    }
+}
+
+tasks.named<AsciidoctorTask>("userguideSinglePageHtml").configure  {
+    val task = this
+    task.extensions.getByType<org.asciidoctor.gradle.jvm.AsciidoctorJExtension>().apply {
+        docExtensions(
+            project.dependencies.create(project(":docs-asciidoctor-extensions")),
+            project.dependencies.create(files("src/main/resources"))
+        )
+    }
+}
```

Comment:
🤔  Now there is quite a bit of duplication. Why was the solution with an if inside configureEach not appropriate?
I'd vote for removing the duplication. We could either have the original solution, or at least extract a common function that configured the extension and call it for the appropriate tasks. I'll leave it up to you to decide, I don't have a strong preference.

Reply:
I thought that's what you meant when you said:
>  could you also make sure the task is explicit?
I misunderstood.

Reply:
Reverting to the old style.